docs/src/tutorial/cython_tutorial.rst

```diff
@@ -165,6 +221,72 @@ which we can try out in the interactive interpreter as follows::
 See, it works! And if you're curious about how much work Cython has saved you,
 take a look at the C code generated for this module.
 
+
+It is always good to check where is the Python interaction in the code with the
+``annotate=True`` parameter in ``cythonize()``. Let's see:
+
+.. figure:: htmlreport.png
+
+The function declaration and return use the Python interpreter so it makes
+sense for those lines to be yellow. Same for the list comprehension because
+it involves the creation of a python object. But the line ``if n % i == 0:``, why?
+We can examine the generated C code to understand:
+
+.. figure:: python_division.png
+
+We can see that some checks happen. Because Cython defaults to the
+Python behavior, the language will perform division checks at runtime,
+just like Python does. You can deactivate those checks by using the
+:ref:`compiler directives<compiler-directives>`.
+
+Now let's see if, even if we have division checks, we obtained a boost in speed.
+Let's write the same program, but Python-style::
+
+    def primes_python(nb_primes):
+        p = []
+        n = 2
+        while len(p) < nb_primes:
+            # Is n prime?
+            for i in p:
+                if n % i == 0:
+                    break
+
+            # If no break occurred in the loop
+            else:
+                p.append(n)
+            n += 1
+        return p
+
+Now we can ensure that those two programs output the same values::
+
+    >>> primes_python(500) == primes(500)
+    True
+
+It's possible to compare the speed now::
+
+    >>> %timeit primes_python(500)
+5.8 ms ± 178 µs per loop (mean ± std. dev. of 7 runs, 100 loops each) ::
+
+    >>> %timeit primes(500)
+502 µs ± 2.22 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
+
+The Cython version is 11 times faster than the Python version! What could explain this?
```

Comment:
11x isn't much, and thus not very convincing. I'm sure that's because 500 is a fairly low number, so the call overhead and the final copying probably still dominate the execution time.

Reply:
I increased that to 1000, but since we can't increase the size of the stack-allocated array, I couldn't get more than that.